File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length;
        int [] ans = new int[n];
        int left =0;
        int right= n-1;
        int p = n-1; // we use this so that we can put the elements from the last index of the ans[] since it is sorted .
        while (left<=right){
            if (Math.abs(nums[left])> Math.abs(nums[right])){
                
                ans[p] = nums[left]* nums[left];
                left ++;
            }
            else{
                ans[p] = nums[right]*nums[right];
                right--;
            }
            p--;
        }
        return ans;

    }
}

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
from typing import List

class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [0] * n     

        left = 0
        right = n - 1
        index_from_last = n - 1

        while left <= right:
            if abs(nums[left]) > abs(nums[right]):
                ans[index_from_last] = nums[left] * nums[left]
                left += 1
            else:
                ans[index_from_last] = nums[right] * nums[right]
                right -= 1

            index_from_last -= 1

        return ans
